src/cargo/ops/cargo_install.rs

```diff
@@ -593,6 +636,20 @@ impl<'gctx> InstallablePackage<'gctx> {
         }
     }
 
+    fn get_rustup_toolchain_source(&self) -> Option<RustupToolchainSource> {
+        self.gctx
+            .get_env("RUSTUP_TOOLCHAIN_SOURCE")
+            .ok()
+            .and_then(|val| match val {
+                "default" => Some(RustupToolchainSource::Default),
+                "env" => Some(RustupToolchainSource::Environment),
+                "cli" => Some(RustupToolchainSource::CommandLine),
+                "path-override" => Some(RustupToolchainSource::OverrideDB),
+                "toolchain-file" => Some(RustupToolchainSource::ToolchainFile),
+                _ => None,
```

Comment:
For forward compatibility, should we have `RustupToolchainSource::Other(String)` and render it verbatim?